accelerate-llvm-native/src/Data/Array/Accelerate/LLVM/Native.hs

```diff
@@ -177,20 +185,22 @@ run1With = runNWith
 -- See also 'runQ', which compiles the Accelerate program at _Haskell_ compile
 -- time, thus eliminating the runtime overhead altogether.
 --
-runN :: Afunction f => f -> AfunctionR f
-runN = runNWith defaultTarget
+runN :: (Afunction f, HasCallStack) => f -> AfunctionR f
+runN = withFrozenCallStack $ runNWith defaultTarget
 
 -- | As 'runN', but execute using the specified target (thread gang).
 --
-runNWith :: forall f. Afunction f => Native -> f -> AfunctionR f
-runNWith target f = go (afunctionRepr @f) afun (return Empty)
+runNWith :: forall f. (Afunction f, HasCallStack) => Native -> f -> AfunctionR f
+runNWith target f
+  = withFrozenCallStack
+  $ go (afunctionRepr @f) afun (return Empty)
```

Comment:
I don't know if it's important, but call stacks won't be frozen in any of the declarations from the where clause. If those need to be frozen, then you need to move them to a let binding.

Reply:
Nah, the call stacks aren't used anywhere (it's there because I was marking where the `run` was called from; that did work but wasn't actually useful so removed it). So the `HasCallStack` constraint is now just left there for when you do your thing (;